src/presentation/http/router/join.ts

```diff
@@ -0,0 +1,57 @@
+import type { FastifyPluginCallback } from 'fastify';
+import type NoteSettingsService from '@domain/service/noteSettings.js';
+import type { TeamMember } from '@domain/entities/team.js';
+
+/**
+ * Represents AI router options
+ */
+interface JoinRouterOptions {
+  /**
+   * Note settings service instance
+   */
+  noteSettings: NoteSettingsService
+}
+
+/**
+ * Join Router plugin
+ *
+ * @param fastify - fastify instance
+ * @param opts - router options
+ * @param done - done callback
+ */
+const JoinRouter: FastifyPluginCallback<JoinRouterOptions> = (fastify, opts, done) => {
+  const noteSettingsService = opts.noteSettings;
+
+  fastify.post<{
+    Params: {
+      hash: string
+    }
+  }>('/:hash', {
+    config: {
+      policy: [
+        'authRequired',
+      ],
+    },
```

Comment:
please, add request validation and response serialisation using schema

Reply:
done